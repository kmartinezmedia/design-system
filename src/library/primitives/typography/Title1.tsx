import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import {
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
} from '@designSystem/theme';

import { Text, TextProps } from './Text';

export const Title1: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 6 },
  ...textProps
}) => {
  return (
    <Text style={styles.title1} spacing={spacing} {...textProps}>
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  title1: {
    fontSize: fontSize.title1,
    lineHeight: lineHeight.title1,
    fontFamily: fontFamily.medium,
    fontWeight: fontWeight.medium,
  },
});
