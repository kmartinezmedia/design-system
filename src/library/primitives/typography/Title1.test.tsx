import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { Title1 } from './Title1';

describe('Title1', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<Title1>Hey, lil 🐛</Title1>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
