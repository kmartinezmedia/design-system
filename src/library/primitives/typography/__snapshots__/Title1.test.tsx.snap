// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title1 default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Medium",
        "fontSize": 28,
        "fontWeight": "500",
        "lineHeight": 40,
      },
      Object {
        "color": "#071422",
        "lineHeight": 40,
        "textAlign": undefined,
      },
      Object {
        "paddingBottom": 24,
      },
      Object {},
      Object {},
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
